services/codeship/codeship.service.js

```diff
@@ -1,19 +1,45 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
+const Joi = require('joi')
+const { BaseSvgScrapingService } = require('..')
+const { isBuildStatus, renderBuildStatusBadge } = require('../build-status')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Codeship extends LegacyService {
+const schema = Joi.object({
+  message: Joi.alternatives()
+    .try(
+      isBuildStatus,
+      Joi.equal('project not found', 'branch not found', 'ignored', 'blocked')
+    )
+    .required(),
+}).required()
+
+const pendingStatus = 'pending'
+const notBuiltStatus = 'not built'
+
+const statusMap = {
+  testing: pendingStatus,
+  waiting: pendingStatus,
+  initiated: pendingStatus,
+  stopped: notBuiltStatus,
+  ignored: notBuiltStatus,
+  blocked: notBuiltStatus,
+  infrastructure_failure: 'failed',
+}
```

Comment:
This mapping seems unusual, though no one has complained about it, so maybe we ought to leave it alone for now.

Reply:
Yeah I know what you mean. This was one of those things that just felt easier to keep consistent with the legacy service experience, at least for the time being